src/main/java/net/neoforged/neoforge/network/filters/GenericPacketSplitter.java

```diff
@@ -207,7 +208,8 @@ public enum RemoteCompatibility {
     }
 
     public static RemoteCompatibility getRemoteCompatibility(Connection manager) {
-        return NetworkRegistry.getInstance().isVanillaConnection(manager) ? RemoteCompatibility.ABSENT : RemoteCompatibility.PRESENT;
+        return NetworkRegistry.getInstance().isConnected(manager, ConnectionPhase.COMMON, SplitPacketPayload.ID) ?
```

Comment:
I think that `COMMON` won't pickup NeoForge-negotiated channels, can you double check? `isConnected` probably needs to check that configuration AND play have the channel, right now it looks like it won't check either with COMMON.

Reply:
No it does, the `ConnectionPhase.isConfiguration()` method is true for `ConnectionPhase.COMMON` and `ConnectionPhase.CONFIGURATION`. While the `ConnectionPhase.isPlay()` returns true for `ConnectionPhase.PLAY` and `ConnectionPhase.COMMON`. So this should work. Testing indicates as such as well.

Reply:
You and your confusing enums... it really doesn't make sense IMO. If a channel is supported in COMMON it should exist for both config and play. (Right now it's an `or` check.)

Since the `ConnectionPhase` is public API this should be addressed I think, at least there. `ConnectionPhase.COMMON.isPlay()` returning true seems a bit off.

Reply:
I have made ConnectionPhase for now internal. We can tweak it and process it in the next few versions